src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.ActionCountTests.cs

```diff
@@ -48,18 +48,81 @@ class Class1
     {
     }
 }";
-            await TestMissingInRegularAndScriptAsync(code);
+            await TestActionCountAsync(code, count: 2);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public async Task MoveType_AvailableOnHeaderIncludingWhitespaceAndAttribute()
+        {
+            var code =
+@"namespace N1
+{[|
+    [X]
+    class Class1
+    {|]
+    }
+}";
+            await TestActionCountAsync(code, count: 2);
         }
 
         [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
-        public async Task MoveType_MissingNotOnHeader3()
+        public async Task MoveType_AvailableAfterHeader()
         {
             var code =
 @"namespace N1
 {
     class Class1
     [||]{
     }
+}";
+            await TestActionCountAsync(code, count: 2);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public async Task MoveType_NotAvailableBeforeType()
+        {
+            var code =
+@"[|namespace N1
+{|]
+    class Class1
+    {
+    }
+}";
+            await TestMissingInRegularAndScriptAsync(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public async Task MoveType_NotAvailableInsideType()
+        {
+            var code =
+@"namespace N1
+{
+    class Class1
+    {[|
+        void M()
+        {
+        }|]
+    }
+}";
+            await TestMissingInRegularAndScriptAsync(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public async Task MoveType_NotAvailableAfterType()
+        {
+            var code =
+@"namespace N1
+{
+    class Class1
+    {
+        void M()
+        {
+        }
+    }
+
+    [|class test1|]
+    {
+    }
 }";
             await TestMissingInRegularAndScriptAsync(code);
```

Comment:
can we have tests showing hte bahvior if inside the attributes for a class, or inside the doc comment for a class?